qt/qt_common/map_widget.cpp

```diff
@@ -246,7 +246,7 @@ void MapWidget::Build()
     fragmentSrc =
         "\
       uniform sampler2D u_sampler; \
-      varying vec2 v_texCoord; \
+      varying highp vec2 v_texCoord; \
```

Comment:
Why did you choose highp here? Should it be chosen according to some dynamic conditions?

Reply:
> Why did you choose highp here? Should it be chosen according to some dynamic conditions?

Qt dinamically changes that by prepending code to the shader program. so if `highp` happens to be unsupported, then it gets turned into `mediump` automatically:
````
#ifndef GL_FRAGMENT_PRECISION_HIGH                                                                                                             
#define highp mediump                                                                                                                          
#endif  
````

Reply:
Oh and since there is no `#version` specification in this shader, then when this is running on a hardware where there is only desktop OpenGL available and the default is `#version 110`, where manual precision specification would normally not be allowed at all, then that case is also covered, because there Qt prepends `define`s that replace the precision keywords with empty tokens:
````
#define lowp
#define mediump
#define highp
````

Reply:
This is clear. What is unclear is _why_ this highp is needed at all here?

Reply:
Because otherwise on actual ES 2.0 HW, where GLSL `#version 100` is selected instead of `#version 110` you don't get a valid shader program. You can check with the official Khronos provided `glslangValidator` validator, that will report `'float' : type requires declaration of default precision qualifier`, which is also what you get in runtime if you try to run this on ES 2.0 HW without the precision qualifier. Qt won't do any magic to make the shader program valid.

Reply:
Thanks for the explanation.

1. Why GLSL 100 can be selected instead of 110? Or is it the case when 110 is not present/not supported in the system?
2. What about other our shaders? Are there any similar missing precision issues?

Reply:
Thank you for the details!